### ISSUE TITLE
QueryBuilder does not add params for Expression in where case

### ISSUE THREAD

**Issue by Tigrov** [yiisoft/yii2#9452]
It does not work if I pass params to new Expression()
```php
AR::find()->where(new Expression(':item = ANY (array)', ['item' => 'value']))->all();
```
but it works if I pass params to where()
```php
AR::find()->where(new Expression(':item = ANY (array)'), ['item' => 'value'])->all();
```
sometimes it is not comfortable when I pass where expression as parameter of function or method
```php
function getAll($where) {
    return AR::find()->where($where)->all();
}

// it does not work
$models = getAll(new Expression(':item = ANY (array)', ['item' => 'value']));
```
This can be fixed if to add one more condition in ```QueryBuilder::buildCondition()```
https://github.com/yiisoft/yii2/blob/master/framework/db/QueryBuilder.php#L910
```php
if ($condition instanceof Expression) {
    foreach ($condition->params as $n => $v) {
        $params[$n] = $v;
    }
    return (string) $condition;
}
```

Also it requires for ```QueryBuilder::buildAndCondition()``` and ```QueryBuilder::buildNotCondition()```
https://github.com/yiisoft/yii2/blob/master/framework/db/QueryBuilder.php#L975
https://github.com/yiisoft/yii2/blob/master/framework/db/QueryBuilder.php#L1004
to replace ```if (is_array($operand)) {``` on ```if (is_array($operand) || $operand instanceof Expression) {```

**Comment by klimov-paul** [comment]
I don't get it: what is the sense of passing `Expression` instance to `Query::where()`, while you can use a plain string for this?

```php
AR::find()->where(':item = ANY (array)', ['item' => 'value'])->all();
```

**Comment by iriscrm** [comment]
This situation described by Tigrov:
"sometimes it is not comfortable when I pass where expression as parameter of function or method

function getAll($where) {
    return AR::find()->where($where)->all();
}"
And I agree with him. Somtimes is userful to prepare where condition dynamically as a single object.
And if Expression using in some methods of QueryBuilder, it is good decision go use it in where(). Isn't it?

**Comment by klimov-paul** [comment]
Resolved by commit bea90e122837cc374dd1ce0199b0a146bfb05e48
